data-prepper-core/src/main/java/org/opensearch/dataprepper/DataPrepper.java

```diff
@@ -90,6 +95,13 @@ public void shutdownDataPrepperServer() {
         dataPrepperServer.stop();
     }
 
+    /**
+     * Triggers shutdown of the Peer Forwarder server.
+     */
+    public void shutdownPeerForwarderServer() {
```

Comment:
We abstract that Data Prepper starts this server, but then expose the shutdown. 

It appears that the current split between `shutdown` and `shutdownDataPrepperServer` is to change the status code returned. Basically, if we shutdown the pipeline, we return 200, even if there is a problem shutting down the Data Prepper Core HTTP Server. What do we want to do when Core Peer Forwarder fails to shutdown? There might be some implications with the buffers here.

Reply:
I don't think there's a different status code for `shutdown` and `shutdownDataPrepperServer`. If shotdwon is successful we return 200 but if it fails we return 500. In both the cases we call `shutdownDataPrepperServer` in finally block

Reply:
Yes, you are correct.

So my question is: What should we use for the behavior of the peer forwarder server? If it fails to shutdown should we return 500 or 200?

Reply:
Status code is dependent on whether we are able to shutdown the pipeline successfully or not.

My thinking was if shutdown call to Data Prepper server doesn't affect the status code, why should shutdown call to Peer Forwarder server affect the return status code?

Reply:
If we need to abstract the shutdown call to Peer Forwarder Server, we can include that in the shutdownDataPrepperServer() method instead of having a separate method.
```
public void shutdownDataPrepperServer() {
    dataPrepperServer.stop();
    peerForwarderServer.stop();
}
```

Reply:
> My thinking was if shutdown call to Data Prepper server doesn't affect the status code, why should shutdown call to Peer Forwarder server affect the return status code?

The only reason I would think would be if it affects the buffers. 

> If we need to abstract the shutdown call to Peer Forwarder Server, we can include that in the shutdownDataPrepperServer() method instead of having a separate method.

I think I agree with this. Maybe we should rename the method as well since it is now shutting down multiple other components.

Reply:
One way to make sure that all the data in buffers are empty is add it as part of `shouldStop()` method in ProcessWorker.
https://github.com/opensearch-project/data-prepper/blob/b02a15e7f95799ed0d664f20d5a5880186e14345/data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/ProcessWorker.java#L80-L87